xwiki-platform-core/xwiki-platform-webjars/xwiki-platform-webjars-api/src/main/java/org/xwiki/webjars/script/WebJarsUrlResolver.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.webjars.script;
+
+import java.util.Map;
+
+import org.xwiki.component.annotation.Role;
+import org.xwiki.stability.Unstable;
+
+/**
+ * Provides operation to make it easy to use WebJars. For instance, it can compute an XWiki WebJars URL.
+ *
+ * @version $Id$
+ * @since 17.3.0RC1
+ */
+@Role
+@Unstable
+public interface WebJarsUrlResolver
```

Comment:
A component which generate URLs does not feel much like a "resolver" to me.

Reply:
Sounds better indeeded, I used `Factory` in  [`b0830af` (#4044)](https://github.com/xwiki/xwiki-platform/pull/4044/commits/b0830af5758dff6efc51ef447f8ec95ec2fd3046)